core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -1693,18 +1694,24 @@ public Status onResult2(final ResolutionResult resolutionResult) {
         return Status.OK;
       }
 
-      List<EquivalentAddressGroup> servers = resolutionResult.getAddresses();
-      channelLogger.log(
-          ChannelLogLevel.DEBUG,
-          "Resolved address: {0}, config={1}",
-          servers,
-          resolutionResult.getAttributes());
-
-      if (lastResolutionState != ResolutionState.SUCCESS) {
-        channelLogger.log(ChannelLogLevel.INFO, "Address resolved: {0}", servers);
-        lastResolutionState = ResolutionState.SUCCESS;
+      StatusOr<List<EquivalentAddressGroup>> serversOrError =
+          resolutionResult.getAddressesOrError();
+      if (serversOrError != null && serversOrError.hasValue()) {
```

Comment:
I suggested above to make serversOrError guaranteed non-null. Things like this become simpler.

Reply:
Done.